robot/Cumulus/tests/browser/gift_entry/lookup_related_fields_validation_for_payment.robot

```diff
@@ -0,0 +1,154 @@
+    &{PAYMENT1} =         API Create Payment            ${OPPORTUNITY}[Id]
+    ...                                  npe01__Payment_Amount__c=100.0
+    ...                                  npe01__Scheduled_Date__c=${DATE}
+    Set suite variable    &{PAYMENT1}
+    &{PAYMENT2} =         API Create Payment            ${OPPORTUNITY}[Id]
+    ...                                  npe01__Payment_Amount__c=300.0
+    ...                                  npe01__Scheduled_Date__c=${DATE}
+    Set suite variable    &{PAYMENT2}
+    ${TEMPLATE_NAME} =    Generate Random String
+	Set suite variable    ${TEMPLATE_NAME}
+    ${BATCH_NAME} =       Generate Random String
+	Set suite variable    ${BATCH_NAME}
+    ${UI_DATE} =          Get Current Date         result_format=%b %-d, %Y
+    Set suite variable    ${UI_DATE}
+    ${NS} =               Get NPSP Namespace Prefix
+    Set suite variable    ${NS}
+
+*** Test Cases ***
+Lookup Related Fields Validation for Payment 
+    [Documentation]          Create template with different default values in payment imported.
+    ...                      Verify  payment related fields are autopopulated in gift entry form.
+    ...                      When batch created and contact is selected, select payment from Review Donations and verifies related values are autopopulated correctly.
+    ...                      Edit the batch and clear out template values, then verify fields are empty.
+    ...                      When a gift is saved verify that the valus are displayed in table row correctly.
+    ...                      When a batch is saved with a gift, open the batch and verify the values are displayed in the form.
+    [tags]                                 unstable      feature:GE        W-8523468
+
+    Go To Page                             Landing                         GE_Gift_Entry
+    Click Link                             Templates
+    Click Gift Entry Button                Create Template
+    Current Page Should Be                 Template                        GE_Gift_Entry
+    Enter Value In Field
+    ...                                    Template Name=${TEMPLATE_NAME}
+    ...                                    Description=This Template is created by automation script
+    Click Gift Entry Button                Next: Form Fields
+    Perform Action On Object Field         select          Payment         Paid         
+    Perform Action On Object Field         select          Payment           Payment Imported
+    Fill Template Form 
+    ...                                    Payment: Paid=&{PAID}                     
+    ...                                    Payment: Check/Reference Number=&{PAYMENT_REF}
+    ...                                    Payment: Payment Method=&{PAYMENT_METHOD}   
+    Load Page Object                       Form         Gift Entry
+    Select Value From lookup               ${PAYMENT1}[Name]                     
```

Comment:
Did "Fill Template Form" keyword not work for you? if it is not I recommend fixing instead

Reply:
okay

Reply:
I looked at the 'Fill Template Form' keyword to take car of Payment Imported Lookup field, it looks like it is not possible to pass payment record ${PAYMENT1}[Name] which created through API as a value like this: &{PAYMENT_IMPORTED}        Default Value=${PAYMENT1}[Name] but it is not working this way. technically it should be like this:&{PAYMENT_IMPORTED}        Default Value=example which will not work for this particular test case because payment is created through API and changing every time when the test case is executed. with that being said, I think using the current keyword is good option. let me know if you have any other suggestion(s)

Reply:
try using "Create Dictionary" keyword from robot builtin library to create the dictionary as part of test data setup like this &{Pmt_imp} =  Create Dictionary      Default Value=${Payment1} and set this &{pmt_imp} as suite variable

Reply:
I tired using 'Create Dictionary' but it is not working with Payment Imported field. It only enters Payment Number into the field instead of selecting from dropdown, as a result the field not get populated with the value and the field will stay blank after saving the form so I won't be able to verify it.

Reply:
I made a fix for this and pushed. lets check builds to make sure its working on metaci as well

Reply:
Great! thank you